subprojects/functional/src/test/groovy/org/gradle/internal/DeferrableTest.groovy

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal
+
+import spock.lang.Specification
+
+import java.util.concurrent.atomic.AtomicInteger
+
+class DeferrableTest extends Specification {
```

Comment:
It seems to me that the `Deferrable` should have some more unit tests, especially for the `map` and `flatMap` methods. WDYT?

Reply:
What are you missing for `flatMap()`? I added some more tests for `map()`.